rider-core/src/main/java/com/github/database/rider/core/dataset/DataSetExecutorImpl.java

```diff
@@ -861,7 +883,90 @@ public void compareCurrentDataSetWith(DataSetConfig expectedDataSetConfig, Strin
 
             DataSetAssertion.assertEqualsIgnoreCols(expectedTable, filteredActualTable, excludeCols);
         }
+    }
+
+    // TODO: refactor
+    private void compareProlog(IDataSet current, IDataSet expected, String[] tableNames) throws DatabaseUnitException {
```

Comment:
Let's move this to a dedicated class and unit test this part? We can create `PrologAssert` in com.github.database.rider.core.assertion package, WDYT?